File: Seccion 12 - Punteros en C++/Seccion_12_Python/10_ejercicio_10.py

nFilas: int
nCols: int
matriz: list = []

def pedir_datos():
    global nFilas, nCols, matriz
    
    nFilas = int(input('Digite el número de filas: '))
    nCols = int(input('Digite el número de columnas: '))
    
    for i in range(nFilas):
        filas: list = []
        for j in range(nCols):
            val = int(input('Digite un número[{}][{}]: '.format(i, j)))
            filas.append(val)
        matriz.append(filas)

def mostrar_matriz():
    print('\nImprimiendo la matriz original:\n')
    for i in range(nFilas):
        for j in range(nCols):
            print('{} '.format(matriz[i][j]), end='')
        print()
    
    print('\nImprimiendo la matriz transpuesta:\n')
    for i in range(nFilas):
        for j in range(nCols):
            print('{} '.format(matriz[j][i]), end='')
        print()

def main():
    
    pedir_datos()
    
    mostrar_matriz()
    
    print(matriz)

if __name__ == "__main__":
    main()

File: Seccion 12 - Punteros en C++/Seccion_12_Dart/10_ejercicio_10.dart

import 'dart:io';

int nFilas, nCols;
List<List> matriz = new List<List>();

void pedirDatos() {

  stdout.write('Digite el número de filas: ');
  nFilas = int.parse(stdin.readLineSync());
  stdout.write('Digite el número de columnas: ');
  nCols = int.parse(stdin.readLineSync());

  for (int i = 0; i < nFilas; i++) {
    List<int> filas = new List<int>();
    for (int j = 0; j < nCols; j++) {
      stdout.write('Digite un número[$i][$j]: ');
      filas.insert(j, int.parse(stdin.readLineSync()));
    }
    matriz.insert(i, new List<int>.from(filas));
    filas.clear();
  }
}

void mostrarMatriz() {
  print('\nImprimiendo la matriz original:');
  for (int i = 0; i < nFilas; i ++) {
    stdout.write('| ');
    for (int j = 0; j < nCols; j++) {
      stdout.write('${matriz[i][j]} ');
    }
    print('|');
  }

  print('\nImprimiendo la matriz transpuesta:');
  for (int i = 0; i < nFilas; i ++) {
    stdout.write('| ');
    for (int j = 0; j < nCols; j++) {
      stdout.write('${matriz[j][i]} ');
    }
    print('|');
  }
}

void main() {

  pedirDatos();

  mostrarMatriz();

  //print(matriz);

}